l10n_ch_import_pain000/models/undo_payment_line.py

```diff
@@ -1,61 +1,119 @@
-
-                account_full_reconcile_id = account_move_line.full_reconcile_id.id
-
-                if account_full_reconcile_id:
-
-                    # Get the counterpart move
-                    account_move_reconcilied = account_move_line_obj.search([
-                                                ('full_reconcile_id', '=',
-                                                account_full_reconcile_id),
-                                                ('move_id', '!=', move_id)])
-
-                    account_move = account_move_obj.search(
-                        [('id', '=', account_move_reconcilied.move_id.id)])
-
-                    account_move_reconcilied.remove_move_reconcile()
-                    # unpost and delete the move from the journal.
-                    account_move.button_cancel()
-                    account_move.unlink()
-
-                # Delete the payment line in the payment order.
-                payment_line.unlink()
-                # Search if there is something left in the payment order.
-                payment_line = payment_line_obj.search([('order_id', '=', move_ids)])
-                if len(payment_line) == 0:
-                    if payment_order_obj.state == 'uploaded':
-                        payment_order_obj.action_done_cancel()
-
-                if display_name:
-                    if data_supp and 'add_tl_inf' in data_supp:
-                        payment_order_obj.message_post(
-                            display_name + " has been removed because : " + data_supp['add_tl_inf'])
-
-                        account_move_line.invoice_id.message_post(
-                            display_name + " has been removed because : " + data_supp['add_tl_inf'])
-
-                    else:
-                        payment_order_obj.message_post(display_name + " has been removed no reason given.")
-                        account_move_line.invoice_id.message_post(display_name + " has been removed no reason given.")
-            return True
-        return False
+    def _post_message(self, data_supp, account_move_lines, payment_order_obj):
+        """
+        This method is used to post message on the invoices that have been
+        deleted from the payment order and it post a message too on the
+        payment order for each deleted invoice.
+
+        :param data_supp: additional information used in the message
+        :param account_move_lines: used to find the invoices where to post
+        the message
+        :param payment_order_obj: used to post the message for each
+        deleted invoices
+        :return: -
+        """
+        if data_supp and 'add_tl_inf' in data_supp:
+
+            for account_move_line in account_move_lines:
+                # Create a link to the invoice that was removed
+                url = '<a href="web#id={}&view_type=form&model=' \
+                      'account.invoice&action=241">{}</a>'. \
```

Comment:
don't hard_code the action in a OCA module. It was fine on our compassion-modules because we only use them but here we shouldn't do that.